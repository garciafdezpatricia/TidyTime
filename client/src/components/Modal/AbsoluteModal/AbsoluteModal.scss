@use "../../../../styles/variables" as *;

.move-task {
    position: absolute;
    top: 30%;
    left: 50%;
    transform: translateX(-50%);
    z-index: 9999;
    width: 15rem;
    display: flex;
    flex-direction: column;
    background-color: $white;
    padding: .8rem;
    border-radius: .4rem;
    box-shadow: 5px 9px 40px -4px rgba(0,0,0,0.75);
    button {
        background-color: transparent;
        border: none;
        outline: none;
        width: fit-content;
        padding-inline: .5rem;
        &:hover {
            background-color: $primary-grey;
        }
    }
}

body.dark-mode .move-task {
    background-color: $background-dark-2;
    color: $white;

    button {
        color: $white;
        &:hover {
            color: $black;
        }
    }
}

.move-button {
    display: flex;
    align-items: center;
    gap: .3rem;
    overflow-x: hidden;
    
    button {
        text-overflow: ellipsis;
        max-width: 13.2rem;
        overflow-x: hidden;
        white-space: nowrap;
    }
}

.move-icon-not-visible {
    display: none;
}

.move-icon-visible {
    display: flex;
}
