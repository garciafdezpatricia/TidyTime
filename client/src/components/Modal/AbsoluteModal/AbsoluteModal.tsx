import { useClickAway } from "@uidotdev/usehooks";
import { useState } from "react";
import { useTranslation } from "react-i18next";
import { TbArrowMoveRight } from "react-icons/tb";
import { uuid } from "uuidv4";

export interface Props {
    options: any[];
    onClick: (arg?:any) => void | any;
    columnIndex: number,
    cardIndex: number,
    onClose: (arg?:any) => void | any;
}


export default function MoveModal({options, onClick, columnIndex, cardIndex, onClose} : Props) {
    const { t } = useTranslation();
    const [iconClass, setIconClass] = useState("move-icon-not-visible");
    const [buttonHovered, setButtonHovered] = useState(-1);
    const [loading, setLoading] = useState(-1);

    const ref = useClickAway(() => {
        onClose();
    })

    const handleHoverOn = (index:number) => {
        setIconClass("move-icon-visible");
        setButtonHovered(index);
    }

    const handleHoverOff = () => {
        setIconClass("move-icon-not-visible");
        setButtonHovered(-1);
    }

    const moving = async (index:number) => {
        setLoading(index);
        await onClick(index);
        setLoading(-1);
    }

    return (
        // @ts-ignore
        <div ref={ref} className="move-task">
            {options.length > 1 ? <p>{t('board.movePanel')}</p> : <p>{t('board.emptyBoard')}</p>}
            {
                options.map((option, index) => {
                    if (index !== columnIndex) {
                        return (
                            <div key={uuid()} className="move-button">
                                <button 
                                    onClick={async () => await moving(index)} 
                                    onMouseOver={() => handleHoverOn(index)}
                                    onMouseOut={() => handleHoverOff()}>
                                    {index === loading && <div className="loader"></div>}
                                    {index === loading && <span>&nbsp;</span>}
                                    {option}
                                </button>
                                <TbArrowMoveRight className={buttonHovered === index ? iconClass : "move-icon-not-visible"} size={"1.2rem"} />
                            </div>
                        )
                    }
                })
            }
        </div>
    )
}