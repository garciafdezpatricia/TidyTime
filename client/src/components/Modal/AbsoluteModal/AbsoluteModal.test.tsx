import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import AbsoluteModal from '@/src/components/Modal/AbsoluteModal/AbsoluteModal';

describe('AbsoluteModal', () => {
    it('renders without crashing', () => {
      const { container } = render(<AbsoluteModal options={[]} onClick={function (arg?: any) {
          throw new Error("Function not implemented.");
      } } columnIndex={0} cardIndex={0} onClose={function (arg?: any) {
          throw new Error("Function not implemented.");
      } }/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<AbsoluteModal options={[]} onClick={function (arg?: any) {
            throw new Error("Function not implemented.");
        } } columnIndex={0} cardIndex={0} onClose={function (arg?: any) {
            throw new Error("Function not implemented.");
        } }/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });