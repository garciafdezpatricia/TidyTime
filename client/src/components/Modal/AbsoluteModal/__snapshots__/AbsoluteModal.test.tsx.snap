// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbsoluteModal matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="move-task"
  >
    <p>
      board.emptyBoard
    </p>
  </div>
</DocumentFragment>
`;
